wire-runtime/src/main/java/com/squareup/wire/Message.java

```diff
@@ -160,46 +161,40 @@ protected void setBuilder(Builder builder) {
     return adapter.fromInt(value);
   }
 
+  /**
+   * Create a {@code byte[]} of the serialized form of the data.
+   * <p>
+   * Note: This is an inefficient means of serializing data. For best performance, call
+   * {@link Wire#adapter} and then {@link MessageAdapter2#write}.
+   */
   @SuppressWarnings("unchecked")
   public byte[] toByteArray() {
```

Comment:
I could be willing to compromise on a new instance method `Wire#toByteArray(Message)` (so as not to duplicate the MessageAdapters). I want a way to serialize that does not require any knowledge of Okio sinks or anything like that as a way in to the API.